File: python/2203.py
while True:
    try:
        xf, yf, xi, yi, v, r1, r2 = map(int, input().split())
        dis = ((xi-xf)**2 + (yi-yf)**2)**0.5
        m = v * 1.5
        dis += m
        efeito = r1+r2
        if dis <= efeito:
            print("Y")
        else:
            print("N")
    except EOFError:
         break

File: dart/2203.dart
import "dart:io";
import "dart:math";

void main() {
  while (true) {
    var l = stdin.readLineSync();

    if (l == null) {
      break;
    }

    List<int> input = l.split(" ").map((e) => int.parse(e)).toList();
    int efect = input[5] + input[6];
    double d = sqrt(pow((input[0] - input[2]), 2) + pow((input[1] - input[3]), 2));
    d += input[4] * 1.5;

    if (d <= efect) {
      print("Y");
    } else {
      print("N");
    }
  }
}
